File: 前端/JieYa/models/banner.js
/**
 * Banner数据模型
 */
export default class Banner {
  constructor(data = {}) {
    this.bannerId = data.bannerId || 0;
    this.image = data.image || '';
    this.type = data.type || 0; // 1-场地 2-商品 3-笔记
    this.on = data.on || 0;
    this.text = data.text || '';
    this.contentId = data.contentId || null;
    this.sort = data.sort || 0;
  }
  
  /**
   * 获取跳转URL
   * @returns {string} 跳转URL
   */
  getJumpUrl() {
    switch (this.type) {
      case 1: // 场地
        return this.contentId ? `/pages/venue/venue-detail?id=${this.contentId}` : '/pages/venue/venue';
      case 2: // 商品
        return this.contentId ? `/pages/prop/prop-detail?id=${this.contentId}` : '/pages/prop/prop';
      case 3: // 笔记
        return this.contentId ? `/pages/find/find-detail?id=${this.contentId}` : '/pages/find/find';
      default:
        return '';
    }
  }
  
  /**
   * 获取类型文本
   * @returns {string} 类型文本
   */
  getTypeText() {
    switch (this.type) {
      case 1:
        return '场地';
      case 2:
        return '商品';
      case 3:
        return '笔记';
      default:
        return '未知';
    }
  }
} 

File: 后端/JieYa-spring/jieya-common/src/main/java/com/example/common/pojo/Banner.java
package com.example.common.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

/**
 * <p>
 * 
 * </p>
 *
 * @author dzk
 * @since 2025-03-20
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("banner")
public class Banner implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(value = "banner_id", type = IdType.AUTO)
    private Integer bannerId;

    private String image;

    /**
     * 1为场地，2为商品，3为笔记
     */
    private Integer type;

    /**
     * 是否启用，1是，0否
     */
    @TableField(value = "`on`")
    private Integer on;

    /**
     * 标题（可选）
     */
    private String text;

    /**
     * 跳转id（如果没有则跳转到整体页面，比如type等于1跳转到场地页面）
     */
    @TableField("content_id")
    private Integer contentId;

    /**
     * 排序号
     */
    private Integer sort;


}
